creator-node/src/services/sync/syncImmediateQueue.js

```diff
@@ -44,6 +44,13 @@ class SyncImmediateQueue {
       connection
     })
 
+    // any leftover active jobs need to be deleted when a new queue
+    // is created since they'll never get processed
+    if (clusterUtils.isThisWorkerInit()) {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      this.deleteOldActiveJobs()
```

Comment:
i think this needs to be a blocking operation otherwise there is a race condition where it is still deleting when the server comes up and starts adding more jobs
@theoilie

Reply:
@jonaylor89 and I synced up offline and thought it should be fine since it fetches active jobs first and then only deletes those. but if it is an issue then we'd have to move this to an init function and await that from serviceRegistry after the constructor (since we can't await in the constructor)

Reply:
I thought about this some more and I think it does need to be blocking or else it'll be a race condition. We can either make an `init()` function of just do `.then()` on the promise. I'll update the PR

Reply:
ah yeah - good callout, it only deletes what is fetched, which is guaranteed to happen before server comes up, so i think this is fine (although looks like you changed already)